src/SME.SGP.Aplicacao/CasosDeUso/PlanoAEE/EncerrarPlanosAEEEstudantesInativosTratarUseCase.cs

```diff
@@ -236,5 +244,54 @@ private bool AlunoFoiTransferidoDaUnidadeEscolar(IEnumerable<AlunoPorTurmaRespos
                     !registroMatriculaTurmaAnterior.CodigoEscola.Equals(registroMatriculaMaisRecente.CodigoEscola))) &&
                    !registroMatriculaTurmaAnterior.CodigoEscola.Equals(registroMatriculaMaisRecente.CodigoEscola);
         }
+
+        private async Task ReaverPlanoEncerradoIndevidamente(PlanoAEE planoAEE, Turma turmaPlano)
+        {
+            try
+            {
+                unitOfWork.IniciarTransacao();
+
+                planoAEE.Situacao = await DefinirSituacaoPlanoAeeReaver(planoAEE, turmaPlano);
+
+                await mediator.Send(new PersistirPlanoAEECommand(planoAEE));
+
+                if (DeveReaverPendencia(planoAEE.Situacao))
+                    await mediator.Send(new ReaverPendenciaPlanoAeeQuery(planoAEE.Id));
+
+                unitOfWork.PersistirTransacao();
+            }
+            catch (Exception ex)
+            {
+                unitOfWork.Rollback();
+                await mediator.Send(new SalvarLogViaRabbitCommand($"Reaver o plano AEE {planoAEE.Id} encerrado indevidamente.", LogNivel.Critico, LogContexto.WorkerRabbit, excecaoInterna: ex.ToString()));
+            }
+        }
+
+        private async Task<SituacaoPlanoAEE> DefinirSituacaoPlanoAeeReaver(PlanoAEE planoAEE, Turma turmaPlano)
+        {
+            if (string.IsNullOrWhiteSpace(planoAEE.ParecerCoordenacao))
+                return await Task.FromResult(SituacaoPlanoAEE.ParecerCP);
+
+            if (!planoAEE.ResponsavelPaaiId.HasValue)
+                return await Task.FromResult(SituacaoPlanoAEE.AtribuicaoPAAI);
+
+            if (string.IsNullOrEmpty(planoAEE.ParecerPAAI))
+                return await Task.FromResult(SituacaoPlanoAEE.ParecerPAAI);
+
+            var periodosEscolares = await mediator
+                .Send(new ObterPeriodosEscolaresPorAnoEModalidadeTurmaQuery(turmaPlano.ModalidadeCodigo, turmaPlano.AnoLetivo, turmaPlano.Semestre));
+
+            var ultimoBimestre = periodosEscolares.LastOrDefault(pe => DateTimeExtension.HorarioBrasilia().Date > pe.PeriodoFim.Date);
```

Comment:
se caso não tiver nenhum período escolar, creio que aqui pode dar erro de referência, não seria válido colocar alguma exceção ou '?' para tratar esse caso?

Reply:
Na linha 286 tem um tratamento se não é nulo, se for já cai na segunda condição do ternário